client/client_test.go

```diff
@@ -0,0 +1,1213 @@
+package client
+
+import (
+	"net"
+	"testing"
+
+	"github.com/gofiber/fiber/v3"
+	"github.com/gofiber/fiber/v3/utils"
+	"github.com/valyala/fasthttp/fasthttputil"
+)
+
+// import (
```

Comment:
please remove the commented out and show the same test cases in the new variant

also add benchmark tests so we can analyze where we can save more allocations and optimize time -> a very important part for fiber(as consumers assume a lot of speed)

Reply:
Yes, I am aware of the problem. And I am adding unit tests and benchmark